--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -4104,6 +4104,8 @@ static int cfg80211_rtw_del_virtual_intf(struct wiphy *wiphy,
                DBG_871X(FUNC_NDEV_FMT" remove monitor interface\n", FUNC_NDEV_ARG(ndev));
        }

+       if (adapter) adapter->pnetdev = NULL;
+
 exit:
        return ret;
 }
--- a/os_dep/linux/os_intfs.c
+++ b/os_dep/linux/os_intfs.c
@@ -1192,8 +1192,10 @@ void rtw_os_ndev_unregister(_adapter *adapter)
        rtw_cfg80211_ndev_res_unregister(adapter);
 #endif

-       if ((adapter->DriverState != DRIVER_DISAPPEAR) && netdev)
+       if ((adapter->DriverState != DRIVER_DISAPPEAR) && netdev) {
                unregister_netdev(netdev); /* will call netdev_close() */
+               adapter->pnetdev = NULL;
+       }

 #if defined(CONFIG_IOCTL_CFG80211) && !defined(RTW_SINGLE_WIPHY)
        rtw_wiphy_unregister(adapter_to_wiphy(adapter));
